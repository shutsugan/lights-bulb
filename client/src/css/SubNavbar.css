.SubNavbar {
	display: flex;
	justify-content: flex-start;
	align-items: center;
	width: 100%;
}

.SubNavbar__button-container {
	margin-right: 16px;
}

.SubNavbar__button--bg-color {
	background: #ebeef3;
}

.SubNavbar__icon--margin {
	margin-right: 8px;
}

.SubNavbar__button--width {
	width: 80px;
}

.SubNavbar__info {
	padding: 0 16px;
	color: #AAA;
	font-size: .8em;
}

h3 {
	color: #3c3e42;
}