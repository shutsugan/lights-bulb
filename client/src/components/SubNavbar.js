import React from 'react';
import { Button } from 'rendition';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faHome } from '@fortawesome/free-solid-svg-icons';
import { faLightbulb } from '@fortawesome/free-solid-svg-icons';
import { faChevronLeft } from '@fortawesome/free-solid-svg-icons';

import '../css/SubNavbar.css';

library.add(faHome);
library.add(faLightbulb);
library.add(faChevronLeft);

const SubNavbar = props => (
	<div className="SubNavbar">
		<div className="SubNavbar__button-container">
			<Button className="SubNavbar__button--bg-color" m={2} square>
				<FontAwesomeIcon icon="home" />
			</Button>
			<Button className="SubNavbar__button--bg-color SubNavbar__button--width">
				<FontAwesomeIcon className="SubNavbar__icon--margin" icon="chevron-left" /> Back
			</Button>
		</div>

		<FontAwesomeIcon className="SubNavbar__icon--margin" icon="lightbulb" />
		<h3>{props.title}</h3>
		<p className="SubNavbar__info">Select a row to interact with it</p>
	</div>
);

export default SubNavbar;