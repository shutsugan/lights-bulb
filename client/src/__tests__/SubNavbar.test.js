import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import SubNavbar from '../components/SubNavbar';

describe('SubNavbar', () => {
    it('should be defined', () => {
      expect(SubNavbar).toBeDefined();
    });
  
    it('should render correctly', () => {
      const tree = shallow(<SubNavbar title="test title" />);
      expect(tree).toMatchSnapshot();
    });

    it('should title be string and equal to passed title', () => {
        const tree = shallow(<SubNavbar name="test title" />);

        expect(typeof(tree.unrendered.props.name)).toBe('string');
        expect(tree.unrendered.props.name).toEqual('test title');
      });
});