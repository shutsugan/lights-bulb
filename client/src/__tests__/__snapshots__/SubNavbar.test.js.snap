// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubNavbar should render correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="SubNavbar"
>
    <div
        className="SubNavbar__button-container"
    >
        <WithTheme(Component)
            className="SubNavbar__button--bg-color"
            m={2}
            square={true}
        >
            <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="home"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                transform={null}
            />
        </WithTheme(Component)>
        <WithTheme(Component)
            className="SubNavbar__button--bg-color SubNavbar__button--width"
        >
            <FontAwesomeIcon
                border={false}
                className="SubNavbar__icon--margin"
                fixedWidth={false}
                flip={null}
                icon="chevron-left"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                transform={null}
            />
             Back
        </WithTheme(Component)>
    </div>
    <FontAwesomeIcon
        border={false}
        className="SubNavbar__icon--margin"
        fixedWidth={false}
        flip={null}
        icon="lightbulb"
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size={null}
        spin={false}
        symbol={false}
        transform={null}
    />
    <h3>
        test title
    </h3>
    <p
        className="SubNavbar__info"
    >
        Select a row to interact with it
    </p>
</div>,
  "nodes": Array [
    <div
      className="SubNavbar"
>
      <div
            className="SubNavbar__button-container"
      >
            <WithTheme(Component)
                  className="SubNavbar__button--bg-color"
                  m={2}
                  square={true}
            >
                  <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon="home"
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        transform={null}
                  />
            </WithTheme(Component)>
            <WithTheme(Component)
                  className="SubNavbar__button--bg-color SubNavbar__button--width"
            >
                  <FontAwesomeIcon
                        border={false}
                        className="SubNavbar__icon--margin"
                        fixedWidth={false}
                        flip={null}
                        icon="chevron-left"
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        transform={null}
                  />
                   Back
            </WithTheme(Component)>
      </div>
      <FontAwesomeIcon
            border={false}
            className="SubNavbar__icon--margin"
            fixedWidth={false}
            flip={null}
            icon="lightbulb"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            transform={null}
      />
      <h3>
            test title
      </h3>
      <p
            className="SubNavbar__info"
      >
            Select a row to interact with it
      </p>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <SubNavbar
      title="test title"
/>,
    "_instance": null,
    "_newState": null,
    "_rendered": <div
      className="SubNavbar"
>
      <div
            className="SubNavbar__button-container"
      >
            <WithTheme(Component)
                  className="SubNavbar__button--bg-color"
                  m={2}
                  square={true}
            >
                  <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon="home"
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        transform={null}
                  />
            </WithTheme(Component)>
            <WithTheme(Component)
                  className="SubNavbar__button--bg-color SubNavbar__button--width"
            >
                  <FontAwesomeIcon
                        border={false}
                        className="SubNavbar__icon--margin"
                        fixedWidth={false}
                        flip={null}
                        icon="chevron-left"
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        transform={null}
                  />
                   Back
            </WithTheme(Component)>
      </div>
      <FontAwesomeIcon
            border={false}
            className="SubNavbar__icon--margin"
            fixedWidth={false}
            flip={null}
            icon="lightbulb"
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            symbol={false}
            transform={null}
      />
      <h3>
            test title
      </h3>
      <p
            className="SubNavbar__info"
      >
            Select a row to interact with it
      </p>
</div>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SubNavbar
    title="test title"
/>,
}
`;
